documentation/user-consent.md

```diff
@@ -48,122 +43,269 @@ The document details aspects regarding the consent management, which includes fo
+    ExpO-->>FE: 302<br>Location: invoker_callback?code=Operatorcode
+  else If Consent is NOT granted - Consent Capture within AuthCode Flow  
+    Note over FE,ExpO: Consent capture  
+    alt when user grants consent / otherwise flow fails  
+      ExpO-->>FE: 302<br>Location: invoker_callback?code=Operatorcode
+    end  
+  end  
+  FE-->>-BE: GET invoker_callback?code=OperatorCode
+  BE->>ExpO: POST /token<br> code=OperatorCode
+  ExpO->>BE: 200 OK <br> {OperatorAccessToken}
+end
+
+BE->>ExpO: Access Operator CAMARA API <br> Authorization: Bearer {OperatorAccessToken}        
+ExpO->>ExpO: Decrypt OperatorAccessToken,<br>grants Access,<br>progresses request to API Backend,<br>gets API response  
+ExpO->>BE: CAMARA API Response
+Note over BE,FE: Response
+```
+
+<br>
+
+**Flow description**:
+
+First, the API invoker (which could potentially be the application BE, an aggregator, etc.) instructs the application frontend in the device to initiate the OIDC authorization code flow with the operator. 
+
+As per the standard authorization code flow, the device application is redirected to the operator authorization endpoint in OGW platform (Steps 1-2), providing a redirect_uri (invoker_callback) pointing to the invoker backend (where the auth code will eventually be sent), as well as the purpose for accessing the data.
+
+The OGW platform receives the request from the device application (Step 3) and does the following:
+
+- Use network based authentication mechanism to obtain the user identifier, i.e.: MSISDN. Set the OAuth sub to the unique user ID in the operator (Step 4).
+
+- Check if user consent is required, which depends on the legal basis associated with the purpose ("legitimate interest", "contract", "consent", etc). If necessary, it will check in the operator's consent master whether user consent has already been given for this identifier and for the requested purpose(s) (Steps 5-6).
+
+Then, two alternatives may occur:
+
+**Scenario 1**: User consent is not required or consent is already given (Step 7). The OGW platform will continue the authorization code flow by redirecting to the API invoker redirect_uri (invoker_callback) and including the authorization code (OperatorCode).
+
+**Scenario 2**: Consent is required and not yet provided by user (Step 8)
+
+- The operator performs the consent capture. Since the authorization code grant involves the frontend, the consent can be captured directly from the user.
+- Once the user has given consent, the authorization code flow continues by redirecting to the API invoker redirect_uri (invoker_callback) and including the authorization code (OperatorCode).
+
+Once the API invoker receives the redirect with the authorization code (OperatorCode - Step 9), it will retrieve the access token from the OGW operator platform (OperatorAccessToken) (Steps 10-11). The OperatorAccessToken issued is encrypted so that no relevant information is exposed.
```

Comment:
```suggestion
Once the API invoker receives the redirect with the authorization code (OperatorCode - Step 9), it will retrieve the access token from the Operator's API Gateway Platform (OperatorAccessToken) (Steps 10-11). The OperatorAccessToken issued is encrypted so that no relevant information is exposed.
```

Reply:
Fixed in https://github.com/camaraproject/IdentityAndConsentManagement/pull/54/commits/c7f9715fa8c597e4c9f94e8f0bbf3aecc501ead5.